packages/asset-uploader/src/helpers/get-asset-path.test.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { Decimal } from "@prisma/client/runtime";
+import { Location } from "@webstudio-is/prisma-client";
+
+const commonAsset = {
+  id: "sa-546",
+  width: new Decimal(200),
+  height: new Decimal(200),
+  projectId: "id",
+  size: 2135,
+  format: "png",
+  createdAt: new Date(),
+  alt: "",
+};
+
+describe("getAssetPath", () => {
+  const OLD_ENV = process.env;
+
+  beforeEach(() => {
+    jest.resetModules();
+    process.env = {
+      ...OLD_ENV,
+      S3_ACCESS_KEY_ID: "testid",
+      S3_SECRET_ACCESS_KEY: "testkey",
+      WORKER_URL: "",
+    };
+  });
+
+  afterAll(() => {
+    process.env = OLD_ENV;
+  });
+
+  describe("local", () => {
+    test("returns local path with no custom path", () => {
+      const { getAssetPath } = require("./get-asset-path");
+      expect(
+        getAssetPath({
+          ...commonAsset,
+          name: "test.png",
+          location: Location.FS,
+        })
+      ).toBe("/uploads/test.png");
+    });
+
+    test("return local path with custom path", () => {
+      process.env.FILE_UPLOAD_PATH = "assets";
+      const { getAssetPath } = require("./get-asset-path");
+      expect(
+        getAssetPath({
+          ...commonAsset,
+          name: "test.png",
+          location: Location.FS,
+        })
+      ).toBe("/assets/test.png");
+    });
+  });
+  describe("s3", () => {
+    test("returns s3 url", () => {
+      process.env.S3_ENDPOINT = "https://fr1.s3.com";
+      process.env.S3_BUCKET = "bucket";
+      process.env.S3_REGION = "fr1";
+      const { getAssetPath } = require("./get-asset-path");
```

Comment:
I would instead just move the functions which access env to the call time, so that you can decide when they access env variables